homeassistant/components/blebox/config_flow.py

```diff
@@ -85,15 +85,14 @@ async def async_step_zeroconf(
         """Handle zeroconf discovery."""
         hass = self.hass
         ipaddress = host_port(discovery_info.__dict__)
-        self.device_config["host"] = discovery_info.host
+        self.device_config["host"] = discovery_info.hostname
```

Comment:
This is, in general, not advised. The problem is that not every network has proper mDNS support, so the IP address is more reliable.

The improvement that can be made in this integration is that the existing config entry is updated (preferably based on the unique ID).

Reply:
@frenck in your proposition you mean if config entry exists and new discovery shows device with same ID, update config data(if IP addresses vary these will be updated?)

Reply:
@frenck [Is that example of what You meant?](https://github.com/home-assistant/core/blob/dev/homeassistant/components/baf/config_flow.py#L60)

Reply:
What I meant is that the hostname should not be used, instead we prefer to use the IP address and update that when needed. We cannot rely the fact that every user is able to resolve the hostname on their network (which we have learned the hard way).

Reply:
Ok, I adjusted accordingly